tests/unit/test_main.py

```diff
@@ -386,11 +388,112 @@ def test_nodeps_check(self):
                         'coordinators, workers, SSH port, and SSH username)'
                         '\n\n' in self.test_stdout.getvalue())
 
-    def test_skip_bad_hosts(self):
+    @patch('prestoadmin.main.load_config', side_effect=mock_load_topology())
+    def test_skip_bad_hosts(self, unused_mock_load):
         main.parse_and_validate_commands(['server', 'install',
                                           "local_path"])
         self.assertEqual(env.skip_bad_hosts, True)
 
+    def test_get_default_options(self):
+        options = Values({'k1': 'dv1', 'k2': 'dv2'})
+        non_default_options = Values({'k2': 'V2', 'k3': 'V3'})
+        default_options = main.get_default_options(options,
+                                                   non_default_options)
+        self.assertEqual(default_options, Values({'k1': 'dv1'}))
+
+    #
+    # The env.port situation is currently a special kind of hell. There are a
```

Comment:
Yup. That is precisely the case. I think the next logical step to config is to move all of the validation to validating the env to somewhere after _update_env finishes and before we kick off tasks. I feel like that's probably a task for after the release though :-)

As a side-effect of doing that work, we can ensure that env.port is set to a consistent type regardless of where it came from.